File: FigurasGo/figuras3d/esfera.go
package figuras3d

import "math"

// Esfera representa una figura tridimensional definida por su radio.
type Esfera struct {
	Radio float64
}

// CalcularVolumen retorna el volumen de la esfera usando la fórmula (4/3) * π * r³.
func (e Esfera) CalcularVolumen() float64 {
	return (4.0 / 3.0) * math.Pi * math.Pow(e.Radio, 3)
}

File: FigurasJava/Figuras3D/Esfera.java
package Figuras3D;



public class Esfera extends Figura3D {
    private double radio;

    // Constructor
    public Esfera(double radio) {
        this.radio = radio;
        this.calcularVolumen();
    }

    // Getter y Setter
    public double getRadio() {
        return radio;
    }

    public void setRadio(double radio) {
        this.radio = radio;
        this.calcularVolumen();
    }

    @Override
    public double calcularVolumen() {
        volumen = (4.0 / 3) * Math.PI * Math.pow(radio, 3);
        return volumen;
    }
}
